File: src/main/java/com/somefrills/events/EntityNamedEvent.java
package com.somefrills.events;

import com.somefrills.misc.Utils;
import net.minecraft.entity.Entity;
import net.minecraft.text.Text;

public class EntityNamedEvent {
    public Entity entity;
    public Text name;
    public String namePlain;

    public EntityNamedEvent(Entity entity, Text name) {
        this.entity = entity;
        this.name = name;
        this.namePlain = Utils.toPlain(name);
    }
}

File: src/main/kotlin/com/somefrills/events/EntityNamedEvent.kt
package com.somefrills.events

import com.somefrills.utils.toPlain
import net.minecraft.world.entity.Entity
import net.minecraft.network.chat.Component

class EntityNamedEvent(val entity: Entity, val name: Component) : FrillsEvent() {
    val namePlain get() = name.toPlain()
}
